File: app/src/main/java/net/treelzebub/zinepress/net/api/model/PocketArticle.java
package net.treelzebub.zinepress.net.api.model;

import com.google.gson.annotations.SerializedName;

/**
 * Created by Tre Murillo on 1/3/16
 */
public class PocketArticle {

    @SerializedName("item_id")
    private final long id;

    @SerializedName("time_added")
    private final long date;

    @SerializedName("resolved_title")
    private final String title;

    @SerializedName("resolved_url")
    private final String url;

    PocketArticle(long id, long date, String title, String url) {
        this.id = id;
        this.date = date;
        this.title = title;
        this.url = url;
    }

    public long getDate() {
        return date;
    }

    public String getTitle() {
        return title;
    }

    public String getUrl() {
        return "https://getpocket/a/read/" + id;
    }

    public long getId() {
        return id;
    }

    public String getOriginalUrl() {
        return url;
    }
}

File: app/src/main/java/net/treelzebub/zinepress/net/api/model/PocketArticle.kt
package net.treelzebub.zinepress.net.api.model

import com.google.gson.annotations.SerializedName
import net.treelzebub.zinepress.db.articles.IArticle

/**
 * Created by Tre Murillo on 2/20/16
 */
class PocketArticle : IArticle {

    @SerializedName("item_id")
    override val id: Long

    @SerializedName("time_added")
    override val date: Long

    @SerializedName("resolved_title")
    override val title: String

    @SerializedName("resolved_url")
    override val originalUrl: String

    val url: String
        get() = "https://getpocket/a/read/$id"

    constructor(id: Long, date: Long, title: String, originalUrl: String) {
        this.id          = id
        this.date        = date
        this.title       = title
        this.originalUrl = originalUrl
    }
}
